### PR TITLE
Android upgrade fixes (tabs => core)

### PR DESCRIPTION
**Pull request by SergeyZhukovsky**
Resolves https://github.com/brave/brave-browser/issues/8456

## Submitter Checklist:

- [ ] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: chromium_src/components/content_settings/core/browser/content_settings_registry.cc [hunk 1]
```diff
@@ -64,6 +64,38 @@
       ContentSettingsInfo::INHERIT_IN_INCOGNITO, \
       ContentSettingsInfo::PERSISTENT, \
       ContentSettingsInfo::EXCEPTIONS_ON_SECURE_AND_INSECURE_ORIGINS); \
+  /* We need 62, 63, 64 for ContentSettingsTypes for Android tabs migration*/ \
```
Comment (bridiver) [inline chromium_src/components/content_settings/core/browser/content_settings_registry.cc]: in brave-core we handle brave-specific content settings by using a content type of plugin and specify the resource id. Even if we did add content setting types, we can't just start at 62 and hope that's will be safe forever

File: browser/android/preferences/brave_pref_service_bridge.cc [hunk 2]
```diff
@@ -175,5 +210,19 @@ jboolean JNI_BravePrefServiceBridge_GetUseRewardsStagingServer(JNIEnv* env) {
       brave_rewards::prefs::kUseRewardsStagingServer);
 }
 
+jboolean JNI_BravePrefServiceBridge_GetBooleanForContentSetting(JNIEnv* env, jint type) {
```
Comment (bridiver) [inline browser/android/preferences/brave_pref_service_bridge.cc]: this should return the java ContentSetting class. It's less confusing than changing types and there are other options for the default like CONTENT_SETTING_ASK

File: chromium_src/components/content_settings/core/browser/content_settings_registry.cc (hunk 1 above)
Comment (bridiver) [reply chromium_src/components/content_settings/core/browser/content_settings_registry.cc]: or is this strictly for reading the old content settings and they will be written to plugins?

File: browser/android/preferences/brave_pref_service_bridge.cc (hunk 2 above)
Comment (SergeyZhukovsky) [reply browser/android/preferences/brave_pref_service_bridge.cc]: we are setting the default values with a new methods, we just need to check bools here. Marking as resolving as discussed that via DM

File: android/java/org/chromium/chrome/browser/upgrade/BraveUpgradeJobIntentService.java [hunk 3]
```diff
@@ -29,12 +32,38 @@
...
+    private static final String PREF_STANDARD_SEARCH_ENGINE_KEYWORD = "brave_standard_search_engine_keyword";
+    private static final String PREF_PRIVATE_SEARCH_ENGINE = "brave_private_search_engine";
+    private static final String PREF_PRIVATE_SEARCH_ENGINE_KEYWORD = "brave_private_search_engine_keyword";
+    private static final String DSE_NAME = "Google";
+    private static final String DSE_KEYWORD = "google.com";
+    // New search engines settings
+    private static final String PRIVATE_DSE_KEYWORD = "brave_private_search_engine_keyword";//"private_dse_keyword";
+    private static final String STANDARD_DSE_KEYWORD = "brave_standard_search_engine_keyword";//"standard_dse_keyword";
+    private static final String PRIVATE_DSE_SHORTNAME = "brave_private_search_engine";//"private_dse_shortname";
+    private static final String STANDARD_DSE_SHORTNAME = "brave_standard_search_engine";//"standard_dse_shortname";
```
[the remark this answers is not in the document]
Comment (SergeyZhukovsky) [reply android/java/org/chromium/chrome/browser/upgrade/BraveUpgradeJobIntentService.java]: No they don't need to be updated there. I probably missed that in code review. We used new pref names in core comparing to tabs repo for some reason :(. And now we need to keep the settings on upgrade. We would avoid that if we used the same names.

File: android/java/org/chromium/chrome/browser/BraveActivity.java [hunk 4]
```diff
@@ -90,6 +92,12 @@ public void onStartWithNative() {
 
         // Disable page prefetching
         PrefetchConfiguration.setPrefetchingEnabledInSettings(false);
+
+        // Set a flag in preferences that tabs pref migration is done
+        SharedPreferences.Editor sharedPreferencesEditor =
+            ContextUtils.getAppSharedPreferences().edit();
+        sharedPreferencesEditor.putBoolean(PREF_TABS_SETTINGS_MIGRATED, true);
+        sharedPreferencesEditor.apply();
```
[the remark this answers is not in the document]
Comment (SergeyZhukovsky) [reply android/java/org/chromium/chrome/browser/BraveActivity.java]: `onSuccess` is related to shields config file migration as well. We check on the file existence there. So I just wanted to put that preference for preferences settings only.

File: android/java/org/chromium/chrome/browser/upgrade/BraveUpgradeJobIntentService.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (SergeyZhukovsky) [reply android/java/org/chromium/chrome/browser/upgrade/BraveUpgradeJobIntentService.java]: Hm, I see what you meant @emerick, yeah my bad as I update the old values to the old, it worked probably in my case as I had them replaced first for the core browser to ensure that works. We should use commented values there. I moved them to a common location and updated.

File: android/java/org/chromium/chrome/browser/BraveActivity.java [hunk 5]
```diff
@@ -70,6 +70,8 @@
     public static final String ANDROID_SETUPWIZARD_PACKAGE_NAME = "com.google.android.setupwizard";
     public static final String ANDROID_PACKAGE_NAME = "android";
     public static final String BRAVE_BLOG_URL = "http://www.brave.com/blog";
+    // Used to indicate were the settings migrated to the new brave-core based version
+    private static final String PREF_TABS_SETTINGS_MIGRATED = "android_tabs_settings_to_core_migrated";
```
[the remark this answers is not in the document]
Comment (SergeyZhukovsky) [reply android/java/org/chromium/chrome/browser/BraveActivity.java]: I moved it to BraveHelper

File: android/java/org/chromium/chrome/browser/BraveActivity.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (emerick) [reply android/java/org/chromium/chrome/browser/BraveActivity.java]: OK, gotcha.

**Pull request merged**